rpcs3/evdev_joystick_handler.cpp

```diff
@@ -251,98 +373,117 @@ std::vector<std::string> evdev_joystick_handler::ListDevices()
+				if (it != devices.end())
+				{
+					libevdev_free(dev);
+					close(fd);
+					return std::distance(devices.begin(), it);
+				}
+
+				// Alright, now that we've confirmed we haven't added this joystick yet, les do dis.
+				devices.push_back({nullptr, fmt::format("/dev/input/%s", et.name), pad });
+				return devices.size() - 1;
+			}
+			libevdev_free(dev);
+			close(fd);
+		}
+	}
+	return -1;
+}
+
 bool evdev_joystick_handler::bindPadToDevice(std::shared_ptr<Pad> pad, const std::string& device)
 {
         Init();
 
-        // Now we need to find the device with the same name, and make sure not to grab any duplicates.
-        fs::dir devdir{"/dev/input/"};
-        fs::dir_entry et;
-        while (devdir.read(et))
-        {
-            // Check if the entry starts with event (a 5-letter word)
-            if (et.name.size() > 5 && et.name.compare(0, 5,"event") == 0)
-            {
-                int fd = open(("/dev/input/" + et.name).c_str(), O_RDONLY|O_NONBLOCK);
-                struct libevdev *dev = NULL;
-                int rc = libevdev_new_from_fd(fd, &dev);
-                if (rc < 0)
-                {
-                    // If it's just a bad file descriptor, don't bother logging, but otherwise, log it.
-                    if (rc != -9)
-                        LOG_WARNING(GENERAL, "Failed to connect to device at %s, the error was: %s", "/dev/input/" + et.name, strerror(-rc));
-                    libevdev_free(dev);
-                    close(fd);
-                    continue;
-                }
-                const std::string name = libevdev_get_name(dev);
-                if (libevdev_has_event_type(dev, EV_KEY) &&
-                    libevdev_has_event_code(dev, EV_ABS, ABS_X) &&
-                    libevdev_has_event_code(dev, EV_ABS, ABS_Y) &&
-                    name == device)
-                {
-                    // It's a joystick.
-
-                    // Now let's make sure we don't already have this one.
-                    bool alreadyIn = false;
-                    for (int i = 0; i < joy_paths.size(); i++)
-                        if (joy_paths[i] == fmt::format("/dev/input/%s", et.name))
-                        {
-                            alreadyIn = true;
-                            break;
-                        }
-                    if (alreadyIn == true)
-                    {
-                        libevdev_free(dev);
-                        close(fd);
-                        continue;
-                    }
-
-                    // Alright, now that we've confirmed we haven't added this joystick yet, les do dis.
-                    joy_paths.emplace_back(fmt::format("/dev/input/%s", et.name));
-                }
-                libevdev_free(dev);
-                close(fd);
-            }
-        }
+				auto find_key = [=](const std::string& name)
```

Comment:
Why capture by `value/copy`? `button_list`, `axis_list` and `rev_axis_list` are rather big maps. Capture by reference `[&]`, maybe list all captured entities explicitly.